### DISCUSSION TITLE
[Request] Standalone Snackbar/Toast component

### DISCUSSION THREAD

**Discussion by douglasg14b** [quasarframework/quasar#9394] [Ideas / Proposals]
Having the plugin is great, however, it seriously lacks the control and flexibility that a component does (Anything the plugin doesn't support or provide behaviors for, cannot be done). Some behaviors may be undesirable for different situations, and we may want to use components inside the toast/snackbar which we can't seem to do?

All in all, the functional approach to it really putting it's use-cases in a tight box.

Can we get an actual component for this? Similar to how Vuetify went about that.

Thoughts?

**Comment by hawkeye64** [comment]
Are you talking about the Notify plugin?

![image](https://user-images.githubusercontent.com/10262924/119409451-8fe8f700-bca4-11eb-9693-634cf8d481db.png)

It follows Material Design's snackbar component here: https://material.io/components/snackbars

What do you think it's lacking?

**Reply by douglasg14b** [reply]
It's lacking the ability to use it like a component, with slots, and all the things that come with that. As a component it's uses are not constrained by the API surface of the plugin. As a plugin, it's use is constrained only to the plugins API surface (Which would have to grow without bounds to cover all use cases a component could cover).

ie: (Ignore nit-picky items, I threw this example together on mobile):

```html
<q-notify :timeout="6000" :active.sync="notificationActive>
   <my-other-component></my-other-component>
   <some-other-thing></some-other-thing> <!-- Can have my own component(s) -->
   <template v-slot:actions="" > <!-- Has named slots that can be used -->
       <q-btn @click="verifyBeforeClose">{{somethingHere}}</q-btn> <!-- Does not close automatically, other logic can exist -->
   </template>
</q-notify
```

As a component, it can be used however the dev wants it to be used. As a plugin, it cannot, think of the  near limitless things you can do with a component, and compare that to the notification API. Things I can't (or afaik I can't) do with it as a plugin:

* **Use my own components, scope, templates, & structure inside of it**
* Control when it's shown or not (Sure, I can spawn a new one, but that's not the same as controlling it from some other state)
* Control when & how it's closed
* Add my own behaviors (ie. Pausing the countdown when hovered, having a menu as an action item...etc endless possibilities)
*  ...etc

**Reply by metalsadman** [reply]
issue a feature request and reference this discussion id.
